src/terms.ml

```diff
@@ -258,8 +261,20 @@ let is_constant : sym -> bool = fun s -> s.sym_prop = Const
 (** [is_private s] tells whether the symbol [s] is private. *)
 let is_private : sym -> bool = fun s -> s.sym_expo = Privat
 
+(** An element of the context. *)
+type hypothesis =
+  | Assume of term Bindlib.var * term
+  (** [Assume(x,a)] assumes that variable [x] is of type [a]. *)
+  | Define of {ctx_v: term Bindlib.var; ctx_y: term; ctx_e: term}
+  (** [Define{ctx_v=x,ctx_y=a,ctx_e=t}] defines variable [x] to be term [t] of
+      type [a]. *)
+
+(** Representation of a typing context, associating a type or term with type
+    to free [Bindlib] variables. *)
+type ctxt = hypothesis list
```

Comment:
Here it would be better to use type `term Bindlib.var * term option * term`, where:
 - `(x, None, a)` represents your `Assume(x, a)`,
 - `(x, Some(t), a)` represents your `Define(x, t, a)` (up to record syntax).

This representation will be more convenient to work with, and it is also more compact in memory (there is one less indirection in the `None` case).